doc/development_process/review_process.rst

```diff
@@ -0,0 +1,215 @@
+- An assignee is responsible for dismissing stale reviews and seeking reviews
+  from additional developers and contributors
+- Pull requests should not be merged without an approval by the assignee.
+
+Pull Request should not be merged by author without review
+===========================================================
+
+All pull requests need to be reviewed and should not be merged by the author
+without a review. The following exceptions apply:
+
+- Hot fixes: Fixing CI issues, reverts, and system breakage
+- Release related changes: Changing version file, applying tags and release
+  related activities without any code changes.
+
+Developers and contributors should always seek review, however there are cases
+when reviewers are not available and there is a need to get a code change into
+the tree as soon as possible.
+
+Reviewers shall not ‘Request Changes’ without comments or justification
+=======================================================================
+
+Any change requests (-1) on a pull request have to be justified. A reviewer
+should avoid blocking a pull-request with no justification. If a reviewer feels
+that a change should not be merged without their review, then: Request change
+of the category: for example:
+
+- Trivial -> Maintainer
+- Assign Pull Request to yourself, this will mean that a pull request should
+  not be merged without your approval.
+
+
+Pull Requests should have at least 2 approvals before they are merged
+======================================================================
+
+A pull-request shall be merged only with two positive reviews (approval). Beside
+the person merging the pull-request (merging != approval), two additional
+approvals are required to be able to merge a pull request. The person merging
+the request can merge without approving or approve and merge to get to the 2
+approvals required.
+
+Reviewers should keep track of pull requests they have provided feedback to
+===========================================================================
+
+If a reviewer has requested changes in a pull request, he or she should monitor
+the state of the pull request and/or respond to mention requests to see if his
+feedback has been addressed. Failing to do so, negative reviews shall be
+dismissed by the assignee or an owner of the repository. Reviews will be
+dismissed following the criteria below:
+
+- The feedback or concerns were visibly addressed by the author
+- The reviewer did not revisit the pull request after 1 week and multiple pings
+  by the author
+- The review is unrelated to the code change or asking for unjustified
+  structural changes such as:
+
+  - Split the PR
+  - Split the commits
+  - Can you fix this unrelated code that happens to appear in the diff
+  - Can you fix unrelated issues
+  - Etc.
+
+Closing Stale Issues and Pull Requests
+=======================================
+
+- The Pull requests and issues sections on Github are NOT discussion forums.
```

Comment:
Some thoughts here:
As it is today, quite a few issues, are indeed discussions. And (when not tagged as bugs) they do provide some benefits (and drawbacks)
"Github issues" do fulfill the role of a discussion medium a bit better than the maillists (less spam, can fix comments, others can follow or ignore at will, the whole thread is keep in one place..).
Though they have the (very) severe issue of their existence not being broadcasted in any way to possible interested parties. Which requires people to constantly monitor the incoming issues.

So maybe we could improve it by being a bit more consistent with issue labels, and have a rule that new discussion issues need to be accompanied with an email to the mail-list ?

Reply:
we do have the right labels in place. This section is just highlighting that issues (bugs, enhancements, etc) need to be driven to a closure and should not be open-ended because of unrelated lengthy discussions or kept open to serve informational purpose.